itou/geiq/notifications.py

```diff
@@ -0,0 +1,44 @@
+from itou.common_apps.address.departments import DEPARTMENT_TO_REGION, REGIONS
+from itou.common_apps.notifications.base_class import BaseNotification
+from itou.communications import NotificationCategory
+from itou.institutions.enums import InstitutionKind
+from itou.institutions.models import InstitutionMembership
+from itou.utils.emails import get_email_message
+
+
+class GEIQImplementationAssessmentSubmittedNotification(BaseNotification):
+    NAME = "geiq_implementation_assessment_submitted"
+    category = NotificationCategory.GEIQ_IMPLEMENTATION_ASSESSMENT
+
+    subject_template = "geiq/email/to_institution_assessment_submitted_subject.txt"
+    body_template = "geiq/email/to_institution_assessment_submitted_body.txt"
+
+    def __init__(self, assessment):
+        self.assessment = assessment
+
+    @property
+    def email(self):
+        relevant_ddets_emails = (
+            InstitutionMembership.objects.filter(
+                institution__department=self.assessment.company.department,
+                institution__kind=InstitutionKind.DDETS_GEIQ,
+                is_active=True,
+            )
+            .select_related("user")
+            .values_list("user__email", flat=True)
+        )
+        relevant_dreets_emails = (
+            InstitutionMembership.objects.filter(
+                institution__department__in=REGIONS[DEPARTMENT_TO_REGION[self.assessment.company.department]],
+                institution__kind=InstitutionKind.DREETS_GEIQ,
+                is_active=True,
+            )
+            .select_related("user")
+            .values_list("user__email", flat=True)
+        )
+
+        to = sorted(set(relevant_ddets_emails) | set(relevant_dreets_emails))
+        context = {
+            "assessment": self.assessment,
+        }
+        return get_email_message(to, context, self.subject_template, self.body_template)
```

Comment:
Dans une version prochaine, il faudra découper les `to` plutôt que d’avoir les mails découpés de manière silencieuse par l’utilitaire d’envoi de mail.
Peut-être envoyer individuellement ?

Reply:
Bah je trouve cela bien dans certain cas (comme celui-ci) que les différents destinataires voient qui est également en copie, cela permet de se répartir le boulot directement.

Reply:
Dans ce cas, je pense qu’il faudrait gérer le cas >= 50 destinataires.

Reply:
Oh pour le moment on est très loin d'avoir 50 mails en combinant des ddets/dreets division GEIQ :sweat_smile: .

Reply:
Donc, c’est OK si :
1. L’email n’est pas envoyé si un destinataire est invalide ? (il me semble bien que c’est le comportement de mailjet)
2. Ça casse le jour où on a 50+ destinataires ?

Reply:
De ma compréhension, ça splitera en 2 emails le jour où il y aura 50+ destinataires non ?
https://github.com/gip-inclusion/les-emplois/blob/master/itou/emails/tasks.py#L47

Reply:
Oui, mais c’est sur le point de changer : https://github.com/gip-inclusion/les-emplois/pull/4180#discussion_r1632561148

Reply:
Héhé on verra à ce moment-là avec le métier ce qu'ils souhaitent pour les différents mails impactés (et celui-ci en fera partie).

Reply:
Ça continue d’empiler du travail pour plus tard :disappointed:

Reply:
Oui mais je pense que la solution choisie sera globalement la même pour la plupart des notifications dans ce cas.
(peut-être juste rajouter une phrase indiquant "Ce message a également été envoyé à Jean Pierre de la DDETS 35 et Nathalie de la DREETS Bretagne" :man_shrugging: ).